src/Generator/Formatting.fs

```diff
@@ -0,0 +1,431 @@
+
+let formatParameterName (name: string) =
+    reservedKeywords
+    |> Array.tryFindIndex (fun n -> name = n)
+    |> Option.map(fun i -> reservedKeywordsUpper |> Array.item i)
+    |> Option.defaultValue name
+
+let formatParam (p: TypedParameterInfo) = formatParameterName p.name
+
+let namespaceForGlSpecification (openGl: RawOpenGLSpecificationDetails) =
+    let prettyName =
+        let temp = openGl.version
+        let versionAsString =
+            let str = openGl.versionNumber |> string
+            str.Replace(".", "")
+        match temp with
+        | StartsWith "GL_ES_VERSION" _
+        | StartsWith "GL_VERSION_ES_CM_1_0" _ -> ".ES" + versionAsString
+        | StartsWith "GL_VERSION" _ -> versionAsString
+        | _ -> temp
+    prettyName
+    |> sprintf "GL%s"
+
+module PrintReady =
+
+    let formatEnumGroup (enumGroup: EnumGroup) =
+        let prettyEnumCases =
+            enumGroup.cases
+            |> Array.Parallel.map(fun case -> 
+                { actualName = case.name
+                  prettyName = case.name |> formatNameRemovingPrefix
+                  value = case.value } : PrintReadyEnum
+            )
+        { groupName = enumGroup.groupName
+          enumCases = prettyEnumCases } : PrintReadyEnumGroup
+
+    let formatTypeInfo (typeInfo: GLType) =
+        { typ = typeInfo
+          prettyTypeName = typeInfo |> typeToString } : PrintReadyTypeInfo
+
+    let formatTypeTypeParameterInfo (typedParameterInfo: TypedParameterInfo) =
+        { actualName = typedParameterInfo.name
+          prettyName = typedParameterInfo |> formatParam
+          typ = typedParameterInfo.typ |> formatTypeInfo } : PrintReadyTypedParameterInfo
+
+    let formatTypedFunctionDeclaration (typedFunctionDeclaration: TypedFunctionDeclaration) =
+        let parameters =
+            typedFunctionDeclaration.parameters
+            |> Array.Parallel.map formatTypeTypeParameterInfo
+        { actualName = typedFunctionDeclaration.name
+          prettyName = typedFunctionDeclaration.name |> formatFunctionName 
+          parameters = parameters
+          retType = typedFunctionDeclaration.retType |> formatTypeInfo } : PrintReadyTypedFunctionDeclaration
+
+let generateDummyTypes =
+    let usings = 
+        [
+            "System"
+        ]
+        
+    let formatDummyType ty =
+        sprintf "public struct %s {}" ty.name
+
+    seq {
+        for using in usings ->
+            writeLine (sprintf "using %s;" using)
+        yield "namespace " + dummyTypesNamespace |> writeLine
+        yield writeLeftBracket
+        yield indent
+        yield writeEmptyLine
+        for ty in additionalTypesToGenerate do
+            match ty._namespace with
+            | Some n ->
+                yield writeLine (sprintf "namespace %s" n)
+                yield writeLeftBracket
+                yield indent
+                yield formatDummyType ty |> writeLine
+                yield unindent
+                yield writeRightBracket
```

Comment:
Shouldn't we recurse here to avoid a potential stack issue?

Reply:
I think everything is fine here. I dont see any possibility for a stack overflow